### PR TITLE
[ADD] ODO-42, commit Avalara transactions in queued job

### PR DESCRIPTION
**Pull request by StefanRijnhart**

### CODE REVIEW

File: avalara_salestax/__openerp__.py [hunk 1]
```diff
@@ -64,8 +64,5 @@
         "images/main_screenshot.png",   
         "images/kranbery_logo.png",     
     ],
-    'demo': [
```
Comment (StefanRijnhart) [inline avalara_salestax/__openerp__.py:67]: Removing the demo config as it messes up tests. Tests only run on init which means the demo config overwrites the custom authentication data in my test database.

File: avalara_salestax/models/account_invoice.py [hunk 2]
```diff
@@ -273,6 +273,33 @@ def test_avalara(self):
             return False, False
         return config, partner
 
+    @api.multi
+    def avalara_commit(self):
```
Comment (StefanRijnhart) [inline avalara_salestax/models/account_invoice.py:277]: Small refactoring to move the actual commit to a distinct method.

File: avalara_salestax/models/avalara_salestax.py [hunk 3]
```diff
@@ -118,7 +118,7 @@ def _get_default_company(self):
         default=True, help="Uncheck the active field to hide the record")
     company_id = fields.Many2one(
         'res.company', 'Company', required=True,
-        default='_get_default_company',
+        default=_get_default_company,
```
Comment (StefanRijnhart) [inline avalara_salestax/models/avalara_salestax.py:121]: Unrelated fix. Noticed default company method was not called because evaluation of quoted methods is only supported for compute methods, not for default methods.

**Review by MPishpecki — APPROVED** [review]
